docs/docs/sample-scripts/entra/find-obsolete-m365-groups/index.mdx

```diff
@@ -0,0 +1,282 @@
+    }
+    else {
+      Write-Host "Following report file will be created in $($Script:Path): 'M365GroupsReport-$($tStamp).csv'."
+      Write-Host "`nAll data will be exported to $($Script:Path): M365GroupsReport-$($tStamp).csv." -ForegroundColor Blue
+      Write-Host "Do not edit this file during the scan." -ForegroundColor Blue
+    }
+    $Script:ReportPath = "$($Script:Path)/M365GroupsReport-$($tStamp).csv"
+  }
+
+  function Get-AllM365Groups {
+    $groups = m365 entra m365group list --includeSiteUrl | ConvertFrom-Json
+    $Script:Groups = $groups | Where-Object { $null -ne $_.siteUrl }
+  }
+
+  function Get-AllGuestUsers {
+    $Script:Guests = m365 entra user list --type Guest | ConvertFrom-Json
+  }
+
+  function Get-AllTeamSites {
+    $Script:TeamSites = m365 spo site list --type TeamSite | ConvertFrom-Json
+  }
+
+  function Start-GroupInsightsTests {
+    Write-Host "Checking $($Script:Groups.Count) groups for activity"
+
+    $Script:Groups | ForEach-Object {
+      $groupInfo = [GroupInfo]::new()
+      $groupInfo.Reference = $_
+      $groupInfo.Membership = @{Owners = 0; Members = 0; Guests = 0 }
+      $groupInfo.TestStatus = "🟢 OK"
+
+      Write-Host "☀︎ $($groupInfo.Reference.displayName)"
+
+      # Tests
+      Test-GroupMembership -Group $groupInfo
+      Test-SharePointActivity -Group $groupInfo
+      Test-ConversationActivity -Group $groupInfo
+
+      # Report
+      New-Report -Group $groupInfo
+    }
+    
+    #Give feedback to user
+    Write-Host "`n-------------------------------------------------------------------"
+    Write-Host "`SUMMARY" -ForegroundColor DarkGreen
+    Write-Host "`-------------------------------------------------------------------"
+    Write-Host "`n👉 Found $($Global:ObsoleteGroups.Count) group$($Global:ObsoleteGroups.Count -gt 1 ? 's' : '') with possibly low activity."
+    Write-Host "`   Please review the report: " -NoNewline
+    Write-Host "$($Script:ReportPath)" -ForegroundColor DarkBlue
+  }
+
+  function Test-GroupMembership {
+    [CmdletBinding()]
+    param (
+      [Parameter(Mandatory = $true)] [GroupInfo] $Group
+    )
+
+    # Original lists
+    $users = m365 entra m365group user list --groupId $Group.Reference.id | ConvertFrom-Json
+    $owners = m365 entra m365group user list --groupId $Group.Reference.id --role Owner | ConvertFrom-Json
+    $members = m365 entra m365group user list --groupId $Group.Reference.id --role Member | ConvertFrom-Json
+    
+    # Consider guests as users that also match the $Script:Guests array
+    $guests = $users | Where-Object { $_.id -in $Script:Guests.id }
+
+    # Modify the $members list to only contain users that are not in the $owners list
+    $members = Compare-Object $members $owners -PassThru
+
+    $Group.Membership = [ordered] @{
+      Owners  = $owners
+      Members = $members
+      Guests  = $guests
+    }
+
+    if ($owners.Count -eq 1 -and ($members.Count + $guests.Count) -eq 0) {
```

Comment:
Why do we ignore groups with no owners? If a group has one owner, and we delete the user from AD, the group will have no owners.

Reply:
Checked.